onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc

```diff
@@ -55,29 +55,85 @@ common::Status SetConvBaseOptions(ModelBuilder& model_builder,
   options.set("dilations", emscripten::val::array(dilations));
   options.set("groups", group);
   // Add Padding.
-  // Usually using autopadding is more efficient than using explicit padding.
-  // Try to see if we can map explicit padding to auto padding.
   std::vector<int64_t> input_shape;
   ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape");
-  AutoPadType auto_pad_type;
-  ORT_RETURN_IF_ERROR(HandleAutoPad(input_shape, weight_shape[2], weight_shape[3],
-                                    helper.Get("pads", std::vector<int64_t>{0, 0, 0, 0}),
-                                    helper.Get("strides", std::vector<int64_t>{1, 1}),
-                                    helper.Get("dilations", std::vector<int64_t>{1, 1}),
-                                    StringToAutoPadType(helper.Get("auto_pad", "NOTSET")),
-                                    auto_pad_type));
-  if (AutoPadType::SAME_UPPER == auto_pad_type || AutoPadType::SAME_LOWER == auto_pad_type) {
-    if (AutoPadType::SAME_LOWER == auto_pad_type) {  // default is SAME_UPPER
-      options.set("autoPad", emscripten::val("same-lower"));
+  AutoPadType auto_pad_type = StringToAutoPadType(helper.Get("auto_pad", "NOTSET"));
+  if (node.OpType() == "Conv") {
+    // Calculate explicit padding for autoPad.
+    if (AutoPadType::SAME_UPPER == auto_pad_type || AutoPadType::SAME_LOWER == auto_pad_type) {
+      std::vector<int64_t> pads_out;
+      ORT_RETURN_IF_ERROR(HandleAutoPad(input_shape, weight_shape[2], weight_shape[3],
```

Comment:
>weight_shape[2], weight_shape[3],

Do we verify somewhere else that the weight shape is at least 4 elements?

Reply:
Ah, checking that is in @honry's CR: https://github.com/microsoft/onnxruntime/pull/18703/files